### PR TITLE
#1 initial structure of adas viewer

### PR DESCRIPTION
**Pull request by opsdep**

### CODE REVIEW

**Comment by opsdep** [comment]
@akorosov @aperrin66 
With this new repo I change the urls of geospaas as below in order to work with both basic viewer and adas viewer:
```python
urlpatterns = [
    url(r'adas/', include('geospaas_adas_viewer.urls')),
    url(r'/', include('geospaas.base_viewer.urls')),
]
```

**Comment by opsdep** [comment]
> @akorosov @aperrin66
> With this new repo I change the urls of geospaas as below in order to work with both basic viewer and adas viewer:
> 
> ```python
> urlpatterns = [
>     url(r'adas/', include('geospaas_adas_viewer.urls')),
>     url(r'/', include('geospaas.base_viewer.urls')),
> ]
> ```
@akorosov @aperrin66 
If this is the correct way, then notice me to push that file in the geospaas repo with a separate branch and pull request and issue

File: geospaas_adas_viewer/views.py [hunk 1]
```diff
@@ -0,0 +1,7 @@
+from geospaas.base_viewer.views import IndexView
+from geospaas_adas_viewer.forms import ADASSearchForm
+
+class adasIndexView(IndexView):
```
Comment (akorosov) [inline geospaas_adas_viewer/views.py]: Class names must be in CamelCase
```suggestion
class AdasIndexView(IndexView):
```

File: geospaas_adas_viewer/urls.py [hunk 2]
```diff
@@ -0,0 +1,9 @@
+from django.conf.urls import url, include
+
+from geospaas_adas_viewer.views import adasIndexView
```
Comment (akorosov) [inline geospaas_adas_viewer/urls.py]:
```suggestion
from geospaas_adas_viewer.views import AdasIndexView
```

File: geospaas_adas_viewer/urls.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+
+
+app_name = 'geospaas_adas_viewer'
+urlpatterns = [
+    url('', adasIndexView.as_view(), name='adasindex'),
```
Comment (akorosov) [inline geospaas_adas_viewer/urls.py]:
```suggestion
    url('', AdasIndexView.as_view(), name='adasindex'),
```

File: geospaas_adas_viewer/tests.py [hunk 4]
```diff
@@ -0,0 +1,125 @@
+from django.test import Client, TestCase
+from django.utils import timezone
+
+from geospaas.base_viewer.tests import BaseViewerHTMLParser
+from geospaas.catalog.models import Dataset
+from geospaas.vocabularies.models import Parameter
+from geospaas_adas_viewer.views import adasIndexView
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]: Please also fix below
```suggestion
from geospaas_adas_viewer.views import AdasIndexView
```

File: geospaas_adas_viewer/forms.py [hunk 5]
```diff
@@ -0,0 +1,19 @@
+from django.contrib.gis import forms
+
+from geospaas.base_viewer.forms import BaseSearchForm
+
+
+class ADASSearchForm(BaseSearchForm):
+    """ Basic version of form for basic seaching of django-geospaas metadata """
```
Comment (akorosov) [inline geospaas_adas_viewer/forms.py]:
```suggestion
    """ Form with extended searching capabilities for ADAS """
```

File: geospaas_adas_viewer/tests.py [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        self.ds = Dataset.objects.all()
+        p1 = Parameter(short_name='dummy',
+                       standard_name='dummy_name', units='dummy')
+        p1.save()
+        self.ds[1].parameters.add(p1)
+        p2 = Parameter(short_name='dummy',
+                       standard_name='name_fake', units='dummy')
+        p2.save()
+        # p2 is added to 'NERSC_test_dataset_titusen' dataset with pk=1
+        self.ds[0].parameters.add(p2)
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]: This should be done in fixtures. Not here.
```suggestion
```

File: geospaas_adas_viewer/tests.py [hunk 7]
```diff
@@ -0,0 +1,125 @@
+from django.test import Client, TestCase
+from django.utils import timezone
+
+from geospaas.base_viewer.tests import BaseViewerHTMLParser
+from geospaas.catalog.models import Dataset
+from geospaas.vocabularies.models import Parameter
+from geospaas_adas_viewer.views import adasIndexView
+
+
+class IntegrationTestsForGUIWithNewBase(TestCase):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]:
```suggestion
class GuiIntegrationTests(TestCase):
```

File: geospaas_adas_viewer/tests.py [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+        p2 = Parameter(short_name='dummy',
+                       standard_name='name_fake', units='dummy')
+        p2.save()
+        # p2 is added to 'NERSC_test_dataset_titusen' dataset with pk=1
+        self.ds[0].parameters.add(p2)
+        self.client = Client()
+        # this BaseViewerHTMLParser is configured to store the desired data of td tag into self.data
+        self.parser = BaseViewerHTMLParser()
+
+    def test_the_post_verb_of_GUI_with_part_of_parameter_name(self):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]:
```suggestion
    def test_post_with_unique_part_of_parameter_name(self):
```

File: geospaas_adas_viewer/tests.py [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+        self.assertEqual(res3.status_code, 200)
+        self.parser.feed(str(res3.content))
+        # both datasets must be in the html
+        self.assertTrue(any([('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data]))
+        # The second one should not be in the html
+        self.assertFalse(any([('file://localhost/some/test/file2.ext' in dat)
+                              for dat in self.parser.data]))
+
+    def test_the_post_verb_of_GUI_with_common_part_of_parameter_name(self):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py:37]:
```suggestion
    def test_post_with_common_part_of_parameter_name(self):
```

File: geospaas_adas_viewer/tests.py [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+            'nameparameters': 'ame'})  # <= Notice the part of both names of 'name_fake' and 'dummy_name' provided by user
+        self.assertEqual(res4.status_code, 200)
+        self.parser.feed(str(res4.content))
+        # both datasets must be in the html
+        self.assertTrue(any([('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data]))
+        self.assertTrue(any([('file://localhost/some/test/file2.ext' in dat)
+                             for dat in self.parser.data]))
+
+    def test_the_get_verb_of_adas_GUI(self):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py:54]:
```suggestion
    def test_get(self):
```

File: geospaas_adas_viewer/tests.py [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+        self.ds[1].parameters.add(p1)
+        p2 = Parameter(short_name='dummy',
+                       standard_name='name_fake', units='dummy')
+        p2.save()
+        # p2 is added to 'NERSC_test_dataset_titusen' dataset with pk=1
+        self.ds[0].parameters.add(p2)
+
+    def tearDown(self):
+        self.form = None
+        self.ds = None
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]: Move to fixtures
```suggestion
```

File: geospaas_adas_viewer/tests.py [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+        self.assertEqual(res5.status_code, 200)
+        self.parser.feed(str(res5.content))
+        # both datasets must be in the html
+        self.assertTrue(any([('file://localhost/some/test/file1.ext' in dat)
+                             for dat in self.parser.data]))
+        self.assertTrue(any([('file://localhost/some/test/file2.ext' in dat)
+                             for dat in self.parser.data]))
+
+
+class FilteringFunctionalityTestsForADASGUI(TestCase):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]:
```suggestion
class ADASSearchFormTests(TestCase):
```

File: geospaas_adas_viewer/tests.py [hunk 13]
```diff
@@ -0,0 +1,125 @@
...
+                       standard_name='name_fake', units='dummy')
+        p2.save()
+        # p2 is added to 'NERSC_test_dataset_titusen' dataset with pk=1
+        self.ds[0].parameters.add(p2)
+
+    def tearDown(self):
+        self.form = None
+        self.ds = None
+
+    def test_filtering_functionality_by_a_fraction_of_parameter_name(self):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]:
```suggestion
    def test_filter_with_unique_part_of_parameter_name(self):
```

File: geospaas_adas_viewer/tests.py [hunk 14]
```diff
@@ -0,0 +1,125 @@
...
+        """shall return the first dataset in fixtures
+        since the 'fak' is set in the name of parameters of
+        the first one """
+        self.form = adasIndexView.form_class({'polygon': '',
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]: Neither form, nor ds should belong to self. Fix below also.
```suggestion
        form = adasIndexView.form_class({'polygon': '',
```

File: geospaas_adas_viewer/tests.py [hunk 15]
```diff
@@ -0,0 +1,125 @@
...
+                                              'source': 1,
+                                              'nameparameters': 'fak'})  # <= Notice the difference betweem tests####
+        self.form.is_valid()
+        self.ds = self.form.filter(self.ds)
+        # one of the datasets should remain as the result of filtering with dummy parameter
+        self.assertEqual(len(self.ds), 1)
+        self.assertEqual(self.ds.first().entry_id,
+                         'NERSC_test_dataset_titusen')
+
+    def test_filtering_functionality_by_similar_part_of_parameter_names(self):
```
Comment (akorosov) [inline geospaas_adas_viewer/tests.py]:
```suggestion
    def test_filter_with_common_part_of_parameter_name(self):
```

**Review by akorosov — CHANGES_REQUESTED** [review]
Very good! I'm glad it takes so few lines to add new functionality!
There are few things to be corrected, mostly cosmetic, see my inline comments. 
Maybe also add a test to ADASSearchFormTests that tests filter which returns no datasets?

File: geospaas_adas_viewer/tests.py (hunk 13 above)
Comment (opsdep) [reply geospaas_adas_viewer/tests.py]: It is better to name it as 

test_filter_with_portion_of_parameter_name

unique does not have the same meaning of fraction and portion
OK?

File: geospaas_adas_viewer/tests.py (hunk 14 above)
Comment (aanersc) [reply geospaas_adas_viewer/tests.py:103]: kindly, for using setUp at the beginning of the test suit, we need to write them with self. OK?

File: geospaas_adas_viewer/tests.py (hunk 13 above)
Comment (akorosov) [reply geospaas_adas_viewer/tests.py]: "unique" alone doesn't, but "unique_part" does have. I guess the main point of the test is that the part of the name 'fak' (what a polysemantic example, btw ;) ) is unique to only one dataset, right? and in another test 'NAM' is common to both datasets. That's why I suggested "unique_part" here and "common_part" there.

**Comment by opsdep** [comment]
@akorosov 
Your mentioned naming is solved in b1eac9b and one more tests is added.

**Review by akorosov — APPROVED** [review]

**Pull request merged**